src/Microdown/MicHTTPResource.class.st

```diff
@@ -7,24 +7,25 @@ Class {
 	#name : #MicHTTPResource,
 	#superclass : #MicAbsoluteResource,
 	#classVars : [
-		'ImageCache'
+		'ImageCache',
+		'ImageNotCachedBlock'
 	],
 	#category : #'Microdown-Core'
 }
 
 { #category : #images }
 MicHTTPResource class >> cachedImageFromUrl: aUrl [
+	"In case a url is not in the cache, it is either read from the net, or if the ImageNotCachedBlock
+	is non nil, the image returned by ImageNotCachedBlock is returned, but not stored in the cache"
 	
-	^ self imageCache 
-		at: aUrl
-		ifAbsentPut: [ (self rawImageFrom: aUrl) ifNil: [ ^ nil ] ] 
-]
-
-{ #category : #accessing }
-MicHTTPResource class >> imageAt: urlString put: aForm [	
-	"the cache contains results of for example ZnEasy getPng:, i.e., a form"
-	
-	ImageCache at: urlString put: aForm
+	(self imageCache includesKey: aUrl)
+		ifTrue: [ ^ self imageCache at: aUrl ].
+		
+	ImageNotCachedBlock 
+		ifNil: [
+			^ self imageCache at: aUrl put: ((self rawImageFrom: aUrl) ifNil: [ ^nil ])]
+		ifNotNil: [ 
+			^ ImageNotCachedBlock cull: aUrl]
```

Comment:
Why do you use cull: and not value: because cull: is costly and reflective so I really tend to use it only when absolutely needed.

Reply:
I believe it was a leftover because I allowed to use a block that could just return a image without using a url - like the box with a "?"

Reply:
I changed it to use value. I had not looked into cull.